package.json

```diff
@@ -27,7 +27,7 @@
     "eslint-config-scality": "scality/Guidelines#8.2.0",
     "eslint-plugin-import": "^2.18.0",
     "mocha": "6.1.4",
-    "typescript": "^4.7.3"
+    "typescript": "^5.3.3"
```

Comment:
typescript 5 is need to generate function overload.
See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-0.html#overload-support-in-jsdoc